src/base64.cpp

```diff
@@ -0,0 +1,52 @@
+#include <migraphx/base64.hpp>
+#include <vector>
+#include <array>
+#include <iostream>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+namespace {
+
+std::array<char, 64> constexpr B64chars{
+    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P',
+    'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f',
+    'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
+    'w', 'x', 'y', 'z', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '+', '/'};
+
+/// base64 encoder snippet altered from https://stackoverflow.com/a/37109258
+const std::string b64_encode(const void* data, const size_t& len)
+{
+    std::vector<char> res_vec((len + 2) / 3 * 4, '=');
+    const unsigned char* p = static_cast<const unsigned char*>(data);
+    size_t j = 0, pad = len % 3;
```

Comment:
Isn't `Padding = 2` when `len == 1`. `Padding = 1` when `len == 2`? 
Example: `w` => `dw==`.       `wo` => `d28=`

Reply:
Pad is not the number of padding in this code

Reply:
I renamed the variable to pad_cond